koku/api/provider/provider_manager.py

```diff
@@ -107,6 +107,51 @@ def get_previous_month_data_exists(self):
             completed_datetime__isnull=False,
         ).exists()
 
+    def get_downloading_state(self):
+        """Get latest manifest downloading state."""
+        state = "pending"
+        manifest = CostUsageReportManifest.objects.filter(
+            provider=self._uuid,
+            billing_period_start_datetime=self.date_helper.this_month_start,
+        ).first()
+        if manifest:
+            if manifest.state.get("download"):
+                if manifest.state.get("download").get("start"):
+                    state = "in-progress"
+                if manifest.state.get("download").get("end"):
+                    state = "complete"
```

Comment:
There's also the potential for a `TypeError` in all the places that do a `get()` chain. Need to default to an empty dict in order to avoid that.

I'm questioning the same translation from `manifest.state` into another value being done in multiple places with a conditional. This seems brittle. A dataclass or [Pydantic model](https://docs.pydantic.dev/2.0/usage/models/) to do the translation would probably be more appropriate.